### PR TITLE
[DI] deprecates tag !tagged in favor of !tagged_iterator

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #31289
| License       | MIT
| Doc PR        | tbd.

### Todo

- [x] fix tests

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ CHANGELOG
  * made `ServiceLocator` implement `ServiceProviderInterface`
  * deprecated support for non-string default env() parameters
  * added `%env(require:...)%` processor to `require()` a PHP file and use the value returned from it
-
+ * deprecated tag `tagged` in favour of `tagged_iterator`
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]: this will be for 4.4 now - please also update the UPGRADE files

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php [hunk 2]
```diff
@@ -117,6 +117,16 @@ function iterator(array $values): IteratorArgument
  * Creates a lazy iterator by tag name.
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: let's add `@deprecated` to force a strike-through in IDEs

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php [hunk 3]
```diff
@@ -117,6 +117,16 @@ function iterator(array $values): IteratorArgument
...
  */
 function tagged(string $tag, string $indexAttribute = null, string $defaultIndexMethod = null): TaggedIteratorArgument
+{
+    @trigger_error('"tagged" tag is deprecated and will be removed in Symfony 5. Please use "tagged_iterator"', E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: Function "%s" is deprecated since Symfony 4.4 [...]

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 4]
```diff
@@ -545,8 +545,14 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
+                case 'tagged_iterator':
                 case 'tagged_locator':
                     $type = $arg->getAttribute('type');
+
+                    if ('tagged' === $type) {
+                        @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated and will be removed in Symfony 5. Please use "tagged_iterator"', $type, $name, $file), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: no need for the "if": this should be moved in the "case" above

File: src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php [hunk 5]
```diff
@@ -724,7 +724,11 @@ private function resolveServices($value, $file, $isParameter = false)
                     throw new InvalidArgumentException(sprintf('"!service_locator" tag only accepts maps of "@service" references in "%s".', $file));
                 }
             }
-            if (\in_array($value->getTag(), ['tagged', 'tagged_locator'], true)) {
+            if (\in_array($value->getTag(), ['tagged', 'tagged_iterator', 'tagged_locator'], true)) {
+                if ('tagged' === $value->getTag()) {
+                    @trigger_error('"!tagged" tag is deprecated and will be removed in Symfony 5. Please use "!tagged_iterator"', E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]: all deprecation template should follow a similar template:
foo is deprecated since Symfony X.Y, use [...] instead.

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 6]
```diff
@@ -16,7 +21,7 @@ CHANGELOG
  * made `ServiceLocator` implement `ServiceProviderInterface`
  * deprecated support for non-string default env() parameters
  * added `%env(require:...)%` processor to `require()` a PHP file and use the value returned from it
-
+
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]: unrelated :)

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md (hunk 1 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/CHANGELOG.md]: Done.

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php (hunk 3 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: Done.

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 7]
```diff
@@ -545,6 +545,8 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
+                case 'tagged_iterator':
+                    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since Symfony 4.4, use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: one case up

File: src/Symfony/Component/DependencyInjection/Loader/schema/dic/services/services-1.0.xsd [hunk 8]
```diff
@@ -265,6 +265,7 @@
       <xsd:enumeration value="iterator" />
       <xsd:enumeration value="service_locator" />
       <xsd:enumeration value="tagged" />
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Loader/schema/dic/services/services-1.0.xsd]: for reference i'd add `<!-- deprecated since Symfony 4.4 -->` here as well

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md (hunk 6 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/CHANGELOG.md]: oh

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php (hunk 4 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: I might be wrong, but wouldn't this mean, that the tag `!tagged` will not work anymore? He will see the deprecation notice, but we force him to use the new tag `!tagged_iterator`.

File: src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php (hunk 5 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]: Done

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php (hunk 7 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: I'm afraid I don't understand. Because if we do this:
```
...
case 'tagged':
    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since Symfony 4.4, use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
    break;
...
```

the `case 'tagged_locator':` wouldn't be reached, right?

Comment (ro0NL) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: see https://3v4l.org/psY8S :)

Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: Yes, leaving out the `break;` would help, but then the fabbot.io check will complain about it.

Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: ok, adding a `// no break` will make fabbot.io happy :-)

Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: Thanks @ro0NL for your help :-)

File: UPGRADE-4.4.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+UPGRADE FROM 4.3 to 4.4
+=======================
+
+DependencyInjection
+-------------------
+
+ * Deprecated tag `tagged` in favour of `tagged_iterator`
```
Comment (ro0NL) [inline UPGRADE-4.4.md]: `in favor of` (same below 2 times)

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md (hunk 6 above)
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]: still unrelated :)

**Review by ro0NL — APPROVED** [review]

File: UPGRADE-4.4.md (hunk 9 above)
Comment (ro0NL) [reply UPGRADE-4.4.md]: `tagged`  is a tag from Yaml perspective, in DI it's just a feature; i suggest

```suggestion
 * Deprecated `tagged` in favor of `tagged_iterator`
```

File: UPGRADE-5.0.md [hunk 10]
```diff
@@ -82,23 +118,6 @@ EventDispatcher
  * The signature of the `EventDispatcherInterface::dispatch()` method has been updated to `dispatch($event, string $eventName = null)`
  * The `Event` class has been removed, use `Symfony\Contracts\EventDispatcher\Event` instead
 
-DependencyInjection
```
Comment (ro0NL) [inline UPGRADE-5.0.md]: if this PR is pending till 4.4; this should be applied in master before 4.3 is branched

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md (hunk 6 above)
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]:
```suggestion

```

File: UPGRADE-5.0.md (hunk 10 above)
Comment (jschaedl) [reply UPGRADE-5.0.md]: Here you go: https://github.com/symfony/symfony/pull/31372 I will revert this change as soon as this PR is merged into master.

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md (hunk 6 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/CHANGELOG.md]: fixed

File: UPGRADE-5.0.md [hunk 11]
```diff
@@ -55,6 +55,42 @@ DependencyInjection
...
+       env(NAME): 1.5
+   ```
+ 
+   After:
+   ```yaml
+   parameters:
+          env(NAME): '1.5'
+   ```
+    
+ * Deprecated tag `tagged` in favour of `tagged_iterator`
```
Comment (jschaedl) [inline UPGRADE-5.0.md]:
```suggestion
 * Removed `tagged` in favour of `tagged_iterator`
```

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 12]
```diff
@@ -16,7 +21,6 @@ CHANGELOG
  * made `ServiceLocator` implement `ServiceProviderInterface`
  * deprecated support for non-string default env() parameters
  * added `%env(require:...)%` processor to `require()` a PHP file and use the value returned from it
-
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]: should be kep I suppose

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php [hunk 13]
```diff
@@ -115,8 +115,20 @@ function iterator(array $values): IteratorArgument
 
 /**
  * Creates a lazy iterator by tag name.
+ *
+ * @deprecated since version 4.4, to be removed in 5.0. Use "tagged_iterator" instead.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: since Symfony [...]

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php [hunk 14]
```diff
@@ -115,8 +115,20 @@ function iterator(array $values): IteratorArgument
...
  */
 function tagged(string $tag, string $indexAttribute = null, string $defaultIndexMethod = null): TaggedIteratorArgument
+{
+    @trigger_error('tagged() is deprecated since version 4.4 and will be removed in 5.0. Use tagged_iterator() instead.', E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: since Symfony [...], use [...]

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: missing namespace btw

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 15]
```diff
@@ -545,6 +545,9 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
+                    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since version 4.4 and will be removed in 5.0. Use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: since Symfony (same everywhere :) )

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php (hunk 13 above)
Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: I just copied this from https://symfony.com/doc/current/contributing/code/conventions.html#deprecations But sure I can change it. Does it make sense to adjust the documentation?

Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]:
```suggestion
 * @deprecated since Symfony 4.4, to be removed in 5.0. Use "tagged_iterator" instead.
```

Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: @nicolas-grekas Like that?

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: Yes, also let's make this a single sentence: ", use"
You're right about the doc, there is a PR open to fix it.

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php (hunk 14 above)
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]:
```suggestion
    @trigger_error(__NAMESPACE__.'\ContainerConfigurator::tagged() is deprecated since Symfony 4.4 and will be removed in 5.0. Use '.__NAMESPACE__.'\ContainerConfigurator::tagged_iterator() instead.', E_USER_DEPRECATED);
```

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php (hunk 15 above)
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]:
```suggestion
                    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since Symfony 4.4 and will be removed in 5.0. Use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php (hunk 13 above)
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]:
```suggestion
 * @deprecated since Symfony 4.4, to be removed in 5.0. Use "tagged_iterator" instead.
```

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 16]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+4.4.0
+-----
+
+ * deprecated tag `tagged` in favor of `tagged_iterator`
```
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]:
```suggestion
 * deprecated `tagged` in favor of `tagged_iterator`
```

File: src/Symfony/Component/DependencyInjection/Loader/schema/dic/services/services-1.0.xsd [hunk 17]
```diff
@@ -264,7 +264,9 @@
       <xsd:enumeration value="constant" />
       <xsd:enumeration value="iterator" />
       <xsd:enumeration value="service_locator" />
+      <!-- "tagged" is deprecated since version 4.4 and will be removed in 5.0. Use "tagged_iterator" instead. -->
```
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/schema/dic/services/services-1.0.xsd]:
```suggestion
      <!-- "tagged" is deprecated since Symfony 4.4 and will be removed in 5.0, use "tagged_iterator" instead. -->
```

File: src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php [hunk 18]
```diff
@@ -724,7 +724,11 @@ private function resolveServices($value, $file, $isParameter = false)
                     throw new InvalidArgumentException(sprintf('"!service_locator" tag only accepts maps of "@service" references in "%s".', $file));
                 }
             }
-            if (\in_array($value->getTag(), ['tagged', 'tagged_locator'], true)) {
+            if (\in_array($value->getTag(), ['tagged', 'tagged_iterator', 'tagged_locator'], true)) {
+                if ('tagged' === $value->getTag()) {
+                    @trigger_error('"!tagged" is deprecated since version 4.4 and will be removed in 5.0. Use "!tagged_iterator" instead.', E_USER_DEPRECATED);
```
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]:
```suggestion
                    @trigger_error('"!tagged" is deprecated since Symfony 4.4 and will be removed in 5.0, use "!tagged_iterator" instead.', E_USER_DEPRECATED);
```

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 19]
```diff
@@ -545,6 +545,9 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
+                    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since Symfony 4.4 and will be removed in 5.0. Use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]:
```suggestion
                    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since Symfony 4.4 and will be removed in 5.0, use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```

File: src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php [hunk 20]
```diff
@@ -115,8 +115,20 @@ function iterator(array $values): IteratorArgument
 
 /**
  * Creates a lazy iterator by tag name.
+ *
+ * @deprecated since Symfony 4.4, to be removed in 5.0, use "tagged_iterator" instead.
  */
 function tagged(string $tag, string $indexAttribute = null, string $defaultIndexMethod = null): TaggedIteratorArgument
+{
+    @trigger_error(__NAMESPACE__.'\ContainerConfigurator::tagged() is deprecated since Symfony 4.4 and will be removed in 5.0, use '.__NAMESPACE__.'\ContainerConfigurator::tagged_iterator() instead.', E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: The `ContainerConfigurator::` part should be removed.

Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]:
```suggestion
    @trigger_error(__NAMESPACE__.'\tagged() is deprecated since Symfony 4.4 and will be removed in 5.0, use '.__NAMESPACE__.'\tagged_iterator() instead.', E_USER_DEPRECATED);
```

Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]:
```suggestion
    @trigger_error(__NAMESPACE__.'\tagged() is deprecated since Symfony 4.4 and will be removed in 5.0, use '.__NAMESPACE__.'\tagged_iterator() instead.', E_USER_DEPRECATED);
```

Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php]: done

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 21]
```diff
@@ -545,6 +545,9 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
+                    @trigger_error(sprintf('Type "%s" of Tag <%s> in "%s" is deprecated since Symfony 4.4 and will be removed in 5.0, use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: s/Tag/tag

**Review by nicolas-grekas — APPROVED** [review]
(with one minor comment)

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php (hunk 21 above)
Comment (jschaedl) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]:
```suggestion
                    @trigger_error(sprintf('Type "%s" of tag <%s> in "%s" is deprecated since Symfony 4.4 and will be removed in 5.0, use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file), E_USER_DEPRECATED);
```

Comment (jschaedl) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: done

**Comment by jschaedl** [comment]
The TravisCi failure seems strange to me. I added the new argument type `tagged_iterator` to the `services-1.0.xsd` (See: https://github.com/symfony/symfony/pull/31321/files#diff-0e535437d820c04db5c34dcc4210e3da) and the XmlParsingException shouldn't occur. Maybe a caching issue?

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml [hunk 22]
```diff
@@ -60,7 +60,7 @@
 
         <!-- transports -->
         <service id="messenger.transport_factory" class="Symfony\Component\Messenger\Transport\TransportFactory">
-            <argument type="tagged" tag="messenger.transport_factory" />
+            <argument type="tagged_iterator" tag="messenger.transport_factory" />
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: this means FWB now needs DI v4.4
this is why tests fail: the composer.json needs an update

**Comment by nicolas-grekas** [comment]
(rebase needed also)

**Comment by jschaedl** [comment]
Rebased and changed target branch to 4.4.

Failed tests seem unrelated.

**Comment by nicolas-grekas** [comment]
The failing job is checking out branch 4.4 and loading dependencies with your patch applied.
This highlights that FrameworkBundle 4.4 is currently not compatible with DI 5.0 in your PR.
That's what needs to be fixed.

**Comment by jschaedl** [comment]
As far as I understand are the tests failing because the FrameworkBundles `services.xml` and `messenger.xml` service configuration files use a xsd file which does not contain the new `tagged_iterator` tag element definition, which was introduced in the current PR: https://github.com/symfony/symfony/pull/31321/files?file-filters%5B%5D=.xsd&owned-by%5B%5D=#diff-0e535437d820c04db5c34dcc4210e3daR269

- services.xml: https://github.com/symfony/symfony/pull/31321/files?file-filters%5B%5D=.xml&owned-by%5B%5D=#diff-9301964a7acd200e8134fed55aede310L5
- messenger.xml: https://github.com/symfony/symfony/pull/31321/files?file-filters%5B%5D=.xml&owned-by%5B%5D=#diff-9750ede1399d486ef37da057a9d613e1L5
- used xsd: https://symfony.com/schema/dic/services/services-1.0.xsd

I have no idea how the `services-1.0.xsd` file under https://symfony.com/schema/dic/services/services-1.0.xsd is generated or how it can be updated. 

@nicolas-grekas Do you have any hints for me?

**Review by xabbuh — APPROVED** [review]
maybe we should keep tests for the legacy behaviour though to make sure that we do not accidentally break it

**Comment by nicolas-grekas** [comment]
Thank you @jschaedl.

**Comment by nicolas-grekas** [comment]
@jschaedl would you mind sending a PR to remove the deprecated codes?

**Comment by jschaedl** [comment]
@nicolas-grekas Sure, I'm happy to do that :-)
